packages/react-core/src/hooks/useDataState.ts

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+interface ActionState<T> {
+  data: T;
+  isLoading: boolean;
+  message: string | undefined;
+}
+
+const getActionState = <T>(data: T): ActionState<T> => ({
+  data,
+  isLoading: false,
+  message: undefined,
+});
+
+export default function useDataState<T, K>(
```

Comment:
Probably be good to put a JSDoc comment here on example usage

Reply:
Will add later 👍